Documentation/project-docs/cross-platform-cryptography.md

```diff
@@ -0,0 +1,241 @@
+Since the system already has a vested interested in making the cryptography libraries safe from security vulnerabilities,
+and already has an update mechanism that system administrators should be using, .NET gets to benefit from this reliability.
+Users who have requirements to use FIPS-validated algorithm implementations also get that benefit for free (when the system
+libraries are FIPS-validated, of course).
+The biggest con is that not all system libraries offer the same capabilities.
+While the core capabilities are present across the various platforms, there are some rough edges.
+
+### Versioning
+
+In .NET Core 1.0 and .NET Core 1.1 the macOS implementation of the cryptography classes was based on OpenSSL.
+In .NET Core 2.0 the dependency was changed to use Apple's Security.framework.
+Within this document "macOS" should use the values for "Linux" if running .NET Core 1.x.
+
+## Hash Algorithms
+
+Hash algorithms, and HMAC algorithms, are very standard bytes-in-bytes-out operations.
+All hash algorithm (and HMAC) classes in .NET Core are deferred to the system libraries (including the \*Managed classes).
+
+While the various system libraries may have different performance, there should not be concerns of compatibility.
+
+In the future there is a possibility that new hash algorithms may be added to .NET Core before one (or more) supported platforms have system support for the algorithm.This would result in a `PlatformNotSupportedException` when invoking the `.Create()` method for the algorithm.
+
+## Symmetric Encryption
+
+The underlying ciphers and chaining are performed by the system libraries.
+
+| Cipher + Mode | Windows | Linux | macOS |
+|---------------|---------|-------|-------|
+| AES-CBC | :white_check_mark: | :white_check_mark: | :white_check_mark: |
+| AES-ECB | :white_check_mark: | :white_check_mark: | :white_check_mark: |
+| 3DES-CBC | :white_check_mark: | :white_check_mark: | :white_check_mark: |
+| 3DES-ECB | :white_check_mark: | :white_check_mark: | :white_check_mark: |
+| DES-CBC | :white_check_mark: | :white_check_mark: | :white_check_mark: |
+| DES-ECB | :white_check_mark: | :white_check_mark: | :white_check_mark: |
+
+In the future there is a possibility that new ciphers may be added to .NET Core before one (or more) supported platforms have system support for it.
+This would result in a `PlatformNotSupportedException` when invoking the `.Create()` method for the algorithm.
+
+In the future there is a possibility that new cipher/chaining modes may be added to .NET Core before one (or more) supported platforms have system support for it.
+This would result in a `PlatformNotSupportedException` when invoking the `.CreateEncryptor()` or `.CreateDecryptor()` methods for the algorithm (or overloads to those methods).
+
+In the future there is a possiblity that new cipher/chaining modes may be added to .NET Core which do not apply to all symmetric algorithms.
```

Comment:
Then how about: `In the future, there is a possibility that new cipher/chaining modes may be added to .NET Core and that these new modes may not apply to all symmetric algorithms.`  Otherwise as the sentence currently reads it suggests that .NET Core does not apply to algorithms, rather than it being the modes that don't.